minisom.py

```diff
@@ -71,39 +71,110 @@ def fast_norm(x):
     return sqrt(dot(x, x.T))
 
 
-def asymptotic_decay(learning_rate, t, max_iter):
-    """Decay function of the learning process.
+def inverse_decay_to_zero(learning_rate, t, max_iter):
+    """Decay function of the learning process that asymptotically
+    approaches zero.
+
     Parameters
     ----------
     learning_rate : float
-        current learning rate.
+        Current learning rate.
 
     t : int
-        current iteration.
+        Current iteration.
 
     max_iter : int
-        maximum number of iterations for the training.
+        Maximum number of iterations for the training.
     """
-    return learning_rate / (1+t/(max_iter/2))
+    C = max_iter / 100.0
+    return learning_rate * C / (C + t)
 
 
-def linear_decay(learning_rate, t, max_iter):
-    return learning_rate * (1 - t/max_iter)
+def linear_decay_to_zero(learning_rate, t, max_iter):
+    """Decay function of the learning process that linearly
+    decreases to zero.
 
+    Parameters
+    ----------
+    learning_rate : float
+        Current learning rate.
 
-def inverse_time_decay(learning_rate, t, max_iter):
-    C = max_iter / 100.0
-    return learning_rate * C / (C+t)
+    t : int
+        Current iteration.
+
+    max_iter : int
+        Maximum number of iterations for the training.
+    """
+    return learning_rate * (1 - t / max_iter)
+
+
+def inverse_decay_to_one(sigma, t, max_iter):
+    """Decay function of sigma that asymptotically approaches one.
+
+    Parameters
+    ----------
+    sigma : float
+        Current sigma.
+
+    t : int
+        Current iteration.
+
+    max_iter : int
+        Maximum number of iterations for the training.
+    """
+    C = (sigma - 1) / max_iter
+    return sigma / (1 + (t * C))
+
+
+def linear_decay_to_one(sigma, t, max_iter):
+    """Decay function of sigma that linearly decreases
+    to one.
+
+    Parameters
+    ----------
+    sigma : float
+        Current sigma.
+
+    t : int
+        Current iteration.
+
+    max_iter : int
+        Maximum number of iterations for the training.
+    """
+    return sigma + (t * (1 - sigma) / max_iter)
+
+
+def asymptotic_decay(dynamic_parameter, t, max_iter):
+    """Legacy default decay function of the learning process
+    and sigma that decays these values asymptotically to 1/3
+    of their original values.
+
+    Using this function may lead to overfitting for sigma values
```

Comment:
I understand that is motivating your changes but I don't agree with this for the following reasons:
- Sigma has a different meaning according to the neighborhood function used.
- Even if you consider a neighborhood gaussian function, a sigma below zero at the end of the training doesn't necessarily results in overfitting if the learning rate decreases appropriately.

Reply:
Good points. Regardless of the neighborhood function used, a sigma value near zero would be updating few surrounding neurons. This could lead to overly localized training where the SOM is training off of what is essentially noise as opposed to the broader trends. However, I agree that there are cases where the learning rate would decrease enough so that this is not an issue. For simplicity, I removed this text along with the corresponding warning message to leave it up to the user to decide whether their SOM is training well.